code/lib/core-server/src/utils/modules-json.test.ts

```diff
@@ -0,0 +1,99 @@
+              type: 'single entry',
+              userRequest:
+                '/Users/ghengeveld/Development/Chromatic/shapes-light/generated-stories-entry.js',
+              loc: 'main[14]',
+            },
+          ],
+          optimizationBailout: ['ModuleConcatenation bailout: Module is not an ECMAScript module'],
+          source:
+            '"use strict";\n\nvar _frameworkImportPath = require("@storybook/react");\n/* eslint-disable import/no-unresolved */\n\n\n(0, _frameworkImportPath.configure)([require.context(\'./src\', true, /^\\.(?:(?:^|\\/|(?:(?:(?!(?:^|\\/)\\.).)*?)\\/)(?!\\.)(?=.)[^/]*?\\.stories\\.mdx)$/), require.context(\'./src\', true, /^\\.(?:(?:^|\\/|(?:(?:(?!(?:^|\\/)\\.).)*?)\\/)(?!\\.)(?=.)[^/]*?\\.stories\\.(js|jsx|ts|tsx))$/)], module, false);',
+        },
+      ])
+    );
+
+    expect(Object.fromEntries(modules)).toEqual({
+      './generated-stories-entry.js': { reasons: new Set() },
+    });
+  });
+
+  it('should trim URL parameters in module identifiers', () => {
+    const modules = webpackStatsToModulesJson(
+      stats([
+        {
+          id: './src/foo.css?ngResource', // note the ?ngResource
+          reasons: [{ moduleId: './src/foo.component.ts' }],
+        },
+        {
+          id: './src/foo.component.ts',
+          reasons: [{ moduleId: './src/foo.stories.ts' }],
+        },
+        {
+          id: './src/foo.stories.ts',
+          reasons: [{ moduleId: CSF_GLOB }],
+        },
+        {
+          id: CSF_GLOB,
+          reasons: [{ moduleId: './.storybook/generated-stories-entry.js' }],
+        },
+      ])
+    );
+
+    expect(Object.fromEntries(modules)).toEqual({
+      './src/foo.css': {
+        reasons: new Set(['./src/foo.component.ts']),
+      },
+      './src/foo.component.ts': {
+        reasons: new Set(['./src/foo.stories.ts']),
+      },
+      './src/foo.stories.ts': {
+        reasons: new Set([CSF_GLOB]),
+      },
+      [CSF_GLOB]: {
+        reasons: new Set(),
+      },
```

Comment:
I thought we were going to handle the logic of "this is a CSF file, included because it matches a `main.js:stories` glob" ourselves?